datalad_next/iter_collections/annexworktree.py

```diff
@@ -0,0 +1,125 @@
+"""Report on the content of a Git-annex repository worktree
+
+The main functionality is provided by the :func:`iter_annexworktree()`
+function.
+"""
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import (
+    Path,
+    PurePath,
+    PurePosixPath,
+)
+from typing import Generator
+
+from more_itertools import intersperse
+
+from .gitworktree import (
+    GitWorktreeItem,
+    GitWorktreeFileSystemItem,
+    iter_gitworktree
+)
+from datalad_next.itertools import (
+    itemize,
+    join_with_list,
+    load_json,
+    route_in,
+    route_out,
+)
+from datalad_next.runners import iter_subproc
+
+lgr = logging.getLogger('datalad.ext.next.iter_collections.annexworktree')
+
+
+# TODO Could be `StrEnum`, came with PY3.11
+class AnnexTreeItemType(Enum):
```

Comment:
This is the same as `GitTreeItemType`. I guess it should derive from that, to get a dedicated type, but not change it or add to it. I think this would be better than an unrelated copy.

Reply:
That would be nice, but enumerations with members, i.e. `GitTreeItemType` cannot be subclassed (see https://docs.python.org/3/howto/enum.html#restricted-enum-subclassing).

Reply:
Ah, true. I wonder if we should even use a separate `enum` then.

Reply:
I removed it